File: Java/LeetC/arrays/PivotIndex.java
package arrays;

class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;

        for (int x : nums) {
            sum += x;
        }

        int leftSum = 0;

        for (int i = 0; i < nums.length; i++) {
            if (leftSum == sum - leftSum - nums[i]) {
                return i;
            }

            leftSum += nums[i];
        }

        return -1;
    }
}

File: python/leetcode/arrays/pivotIndex.py
from typing import List

def pivotIndex(nums: List[int]) -> int:
    S = sum(nums)
    leftSum = 0

    for i, x in enumerate(nums):
        if leftSum == (S - leftSum -  x):
            return i
        leftSum += x
    return -1


if __name__ == "__main__":
    a = [1, 7, 3, 6, 5, 6]
    r = pivotIndex(a)
    print(r)
